test/exe/Main.hs

```diff
@@ -1489,6 +1512,25 @@ openTestDataDoc path = do
   source <- liftIO $ readFileUtf8 $ "test/data" </> path
   openDoc' path "haskell" source
 
+findCodeActions :: TextDocumentIdentifier -> Range -> [T.Text] -> Session [CodeAction]
+findCodeActions doc range expectedTitles = do
+  actions <- getCodeActions doc range
+  let matches = sequence
+        [ listToMaybe
+          [ action
+          | CACodeAction action@CodeAction { _title = actionTitle } <- actions
+          , actionTitle == expectedTitle ]
+        | expectedTitle <- expectedTitles]
+  let msg = show $
+            [ actionTitle
+            | CACodeAction CodeAction { _title = actionTitle } <- actions
+            ]
+  liftIO $ assertBool msg (isJust matches)
```

Comment:
This slightly changes the meaning of some of the tests. Before they were checking that the actual list of actions equals the expected list of actions. This seems to check that the actual list of actions is a super set of the expected actions. Also the assertion message now only shows the actual actions and not the expected actions.

Why not keep checking for equality with `@=?`?

Reply:
The change is motivated by the list of suggested code actions now including actions to introduce new definitions. I assume that this test doesn't care about those extra actions, and therefore enumerating them here, in order to be able to use `@=?`, did not seem appropriate. 

Thanks for the suggestion to improve the error message, I'll take a look at that.